sbi/inference/snre/snre_base.py

```diff
@@ -273,7 +272,13 @@ def _train(
         # If we're retraining from scratch each round, reset the neural posterior
         # to the untrained copy we made at the start.
         if self._retrain_from_scratch_each_round:
-            self._posterior = deepcopy(self._posterior)
+            self._posterior = deepcopy(self._untrained_posterior)
+            self._summary_net = deepcopy(self._untrained_summary_net)
+            optimizer = optim.Adam(
+                list(self._posterior.net.parameters())
+                + list(self._summary_net.parameters()),
```

Comment:
Since the `Identity` module should not have trainable parameters, it would be safe to always have a summary net and just overwrite it when one is provided.

Reply:
In fact, could we make the default `summary_net` `Identity()` instead of None? It is not mutable and hence it should cause no problems.

Reply:
I don't get it. Are you suggesting to make an if-case right here in line 278? If so, I do not think that this makes the code more legible.